client/state/notifications/index.js

```diff
@@ -0,0 +1,14 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { NOTIFICATIONS_UNSEEN_COUNT_SET } from 'state/action-types';
+
+export const unseenCount = ( state = 0, action ) =>
+	NOTIFICATIONS_UNSEEN_COUNT_SET === action.type ? action.unseenCount : state;
+
+export const setUnseenCount = count => ( {
```

Comment:
Maybe export this as `state.notifications.count` (if we think we'll want other state) or `state.notificationsUnseenCount` if we just want a single count.